scripts/Diffraction/isis_powder/pearl.py

```diff
@@ -181,10 +185,21 @@ def _output_focused_ws(self, processed_spectra, run_details, output_mode=None):
         if not output_mode:
             output_mode = self._inst_settings.focus_mode
 
+        attenuation_path = None
         if self._inst_settings.perform_atten:
-            attenuation_path = self._inst_settings.attenuation_file_path
-        else:
-            attenuation_path = None
+            name_key='name'
+            path_key='path'
+            if isinstance(self._inst_settings.attenuation_files, str):
+                self._inst_settings.attenuation_files = eval(self._inst_settings.attenuation_files)
+            for atten_file in self._inst_settings.attenuation_files:
+                if any (required_key not in atten_file for required_key in [name_key,path_key]):
+                    logger.warning("A dictionary in attenuation_files has been ignored because "
+                                   "it doesn't contain both {} and {} entries".format(name_key,path_key))
```

Comment:
This is really pedantic and I sometimes find myself using the older .format syntax too, but maybe it's better to use an f-string?

Reply:
No problem, will be bit more concise. Have just pushed a change for this line and a similar line a few lines lower